proposals/stdlib/long-ariphmetics.md

```diff
@@ -0,0 +1,155 @@
+
+# Extending Kotlin API for BigInteger and BigDecimal
+
+* **Type**: Standard Library API proposal
+* **Author**: Daniil Vodopian
+* **Contributors**: Ilya Gorbunov
+* **Status**: Submitted
+* **Prototype**: Not started
+
+
+## Summary
+
+Overload mathematical operations and infix functions to work with BigInteger and BigDecimal.
+
+## Similar API review
+
+* Python's `long` type: https://docs.python.org/2/library/stdtypes.html#numeric-types-int-float-long-complex
+
+## Use cases and motivation
+
+The motivation is to provide a complete and symmetrical set of operators for long arithmetics provided by JDK, namely `BigInteger` for integer computations and `BigDecimal` for floating point computations. 
+
+A "complete" set of operations in this context means that all operators on a basic type are available on the corresponding big type. Operations for `BigInteger` are modelled after `Long`, and operations on `BigDecimal` are modelled after `Double`. The goal is to improve the Java API and not to introduce any additional semantics.
+
+Kotlin stdlib already [contains](https://github.com/JetBrains/kotlin/blob/master/libraries/stdlib/src/kotlin/util/BigNumbers.kt) 12 of 43 proposed functions, but that set is neither complete nor symmetrical. That determines the placement of the additional functions.  
+
+The proposal does not include mixed operations between `BigInteger` and `BigDecimal` since we did not find any evidence of their use.
+
+## Alternatives
+
+* The JDK API for operations on `BigInteger` and `BigDecimal`
+* Converting basic numeric types to and from "big" types like `1.toBigInteger()` or `2.toBig()` or possibly `3.big`
+
+## Dependencies
+
+* JDK6 `BigInteger` and `BigDecimal`
+
+## Placement
+
+* Standard Library, `java.math`, since there is already a part of the prosed API for [BigInteger](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/java.math.-big-integer/) and [BigDecimal](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/java.math.-big-decimal/)
```

Comment:
Currently implemented part of API is in `kotlin` package: https://github.com/JetBrains/kotlin/blob/master/libraries/stdlib/src/kotlin/util/BigNumbers.kt